### DISCUSSION TITLE
Is there a better way to query many-many data?

### DISCUSSION THREAD

**Discussion by gcostaapps** [supabase/supabase#3859] [Questions] [answered]
Hi, I'm having troubles to get many-many data in the correct structure. Is there a better way to get the data than what I'm doing?

I have 4 tables, class, users, class_teacher, where class_teacher is a join table between class and users who have a role of teacher.

I'm querying this way:
```
final query = _client.from('class').select('''
    *,
    teachers:class_teacher(
      teacher:teacher_id(*)
      )
    ''')
```

I was hoping to get the data like this:

```
[{
   ...userData
  teachers: [
      {...teacherData},
      {...teacherData},
   ]
}...]
```

But instead I'm getting

```
[{
   ...userData
  teachers: [
      {teacher: {...teacherData}},
   ]
}...]
```

I could not understand if it's the same problem of [this comment](https://github.com/supabase/supabase/discussions/1291#discussioncomment-1535063)

**Comment by steve-chavez** [comment, marked as answer]
Have you tried doing:

```js
final query = _client.from('class').select('''
    *,
    teachers:users(*)
    ''')
```

**Reply by gcostaapps** [reply]
Yes, I got an error saying "Could not find a relationship between class and users in the schema cache".

**Reply by steve-chavez** [reply]
Can you share the `details` key in the error?

**Reply by gcostaapps** [reply]
The details property as well as the code is `null`, the hint field is `"If a new foreign key between these entities was created in the database, try reloading the schema cache."`

**Reply by gcostaapps** [reply]
Sorry, my database table is named `user` and not `users`, I fixed it in the query that you sent and everything works now. Great to see that it works this way! Thank you for the suppport!!
